spring-integration-ip/src/main/java/org/springframework/integration/ip/dsl/Udp.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.dsl;
+
+/**
+ * Factory methods for UDP.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public final class Udp {
+
+	private Udp() {
+		super();
+	}
+
+	public static UdpInboundChannelAdapterSpec inboundAdapter(int port) {
+		return new UdpInboundChannelAdapterSpec(port);
+	}
+
+	public static UdpInboundChannelAdapterSpec inboundMulticastAdapter(int port, String multicastGroup) {
+		return new UdpInboundChannelAdapterSpec(port, multicastGroup);
+	}
+
+	public static UdpOutboundChannelAdapterSpec outboundAdapter(String destinationExpression) {
+		return new UdpOutboundChannelAdapterSpec(destinationExpression);
+	}
+
+	public static UdpOutboundChannelAdapterSpec outboundMulticastAdapter(String destinationExpression) {
+		return new UdpMulticastOutboundChannelAdapterSpec(destinationExpression);
```

Comment:
I think the return type of the method must be `UdpMulticastOutboundChannelAdapterSpec` as well.
Otherwise we don't have access to the `timeToLive()`  method from the target code.